_blog/2018/soft-multitenancy.md

```diff
@@ -0,0 +1,298 @@
+istio-ingress-68d65fc5c6-dnvfl            1/1       Running   0          1d
+istio-mixer-5b9f8dffb5-8875r              3/3       Running   0          1d
+istio-pilot-678fc976c8-b8tv6              2/2       Running   0          1d
+istio-sidecar-injector-7587bd559d-5tgk6   1/1       Running   0          1d
+prometheus-cf8456855-hdcq7                1/1       Running   0          1d
+servicegraph-75ff8f7c95-wcjs7             1/1       Running   0          1d
+```
+However, accessing all the cluster's pods or pods from another tenant's namespace is not:
+```sh
+dev-admin:~/go/src/istio.io/istio# kubectl get pods --all-namespaces
+Error from server (Forbidden): pods is forbidden: User "dev-admin" cannot list pods at the cluster scope
+dev-admin:~/go/src/istio.io/istio# kubectl get pods -n istio-system1
+Error from server (Forbidden): pods is forbidden: User "dev-admin" cannot list pods in the namespace "istio-system1"
+
+```
+
+The tenant administrator can deploy applications in the application namespace configured for
+that tenant. As an example, updating the [bookinfo](https://istio.io/docs/guides/bookinfo.html)
+manifests and then deploying under the tenant's application namespace of *ns-0*, listing the
+pods in use by this tenant's namespace is permitted:
+```sh
+dev-admin:~/go/src/istio.io/istio# kubectl get pods -n ns-0
+NAME                              READY     STATUS    RESTARTS   AGE
+details-v1-64b86cd49-b7rkr        2/2       Running   0          1d
+productpage-v1-84f77f8747-rf2mt   2/2       Running   0          1d
+ratings-v1-5f46655b57-5b4c5       2/2       Running   0          1d
+reviews-v1-ff6bdb95b-pm5lb        2/2       Running   0          1d
+reviews-v2-5799558d68-b989t       2/2       Running   0          1d
+reviews-v3-58ff7d665b-lw5j9       2/2       Running   0          1d
+
+```
+
+But accessing another tenant's application namespace is not:
+```sh
+dev-admin:~/go/src/istio.io/istio# kubectl get pods -n ns-1
+Error from server (Forbidden): pods is forbidden: User "dev-admin" cannot list pods in the namespace "ns-1"
+
+```
+
+If the [addon tools](https://istio.io/docs/tasks/telemetry/), example
+[prometheus](https://istio.io/docs/tasks/telemetry//querying-metrics.html), are deployed
+(also limited by an Istio `namespace`) the statistical results returned would represent only
+that traffic seen from that tenant's application namespace.
+
+## Conclusion
+The evaluation performed indicates Istio has sufficient capabilities and security to meet a
+small number of multitenant use cases. It also shows that Istio and Kubernetes can __not__
+provide sufficient capabilities and security for many other use cases.  Especially use
+cases that require complete security and isolation between the tenants that may be
+malicious towards each other. The improvements required to reach this next level of security
+and isolation are more in Kubernetes or container technology than improvements in Istio capabilities.
```

Comment:
Also reads awkwardly.  Maybe -> Improvements required to reach a more secure model of security and isolation require work in Kubernetes and the underlying CRI implementaiton, rather than improvements in Istio capabilities.